src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -274,27 +274,51 @@ public class ReleaseService : IReleaseService
                 .OnSuccess(_mapper.Map<ReleasePublicationStatusViewModel>);
         }
 
-        public async Task<Either<ActionResult, ReleaseViewModel>> EditReleaseSummaryAsync(
-            Guid releaseId, UpdateReleaseSummaryRequest request)
+        public async Task<Either<ActionResult, ReleaseViewModel>> UpdateReleaseAsync(
+            Guid releaseId, UpdateReleaseRequest request)
         {
             return await _persistenceHelper
                 .CheckEntityExists<Release>(releaseId)
                 .OnSuccess(_userService.CheckCanUpdateRelease)
-                .OnSuccess(_ => ValidateReleaseSlugUniqueToPublication(request.Slug, releaseId, releaseId))
-                .OnSuccess(async () =>
+                .OnSuccess(release => _userService.CheckCanUpdateReleaseStatus(release, request.Status))
+                .OnSuccessDo(() => ValidateReleaseSlugUniqueToPublication(request.Slug, releaseId, releaseId))
+                .OnSuccessDo(() => CheckAllDatafilesUploadedComplete(releaseId, request.Status))
+                .OnSuccess(async release =>
                 {
-                    var release = await EntityFrameworkQueryableExtensions.FirstOrDefaultAsync(_context.Releases
-                            .Where(r => r.Id == releaseId));
+                    if (request.Status == ReleaseStatus.Approved &&
+                        request.PublishMethod == PublishMethod.Scheduled &&
+                        !request.PublishScheduledDate.HasValue)
+                    {
+                        return ValidationActionResult(ApprovedReleaseMustHavePublishScheduledDate);
+                    }
 
                     release.Slug = request.Slug;
                     release.TypeId = request.TypeId;
-                    release.PublishScheduled = request.PublishScheduledDate;
                     release.ReleaseName = request.ReleaseName;
-                    release.NextReleaseDate = request.NextReleaseDate;
                     release.TimePeriodCoverage = request.TimePeriodCoverage;
+
+                    var isApproved = release.Status == ReleaseStatus.Approved || request.Status == ReleaseStatus.Approved;
+
+                    release.Status = request.Status;
+                    release.InternalReleaseNote = request.InternalReleaseNote;
+                    release.NextReleaseDate = request.NextReleaseDate;
+
+                    release.PublishScheduled = request.PublishMethod == PublishMethod.Immediate && isApproved
+                        ? DateTime.UtcNow
+                        : request.PublishScheduledDate;
+
+                    _context.Releases.Update(release);
+                    await _context.SaveChangesAsync();
+
+                    // Only need to inform Publisher if changing release status to or from Approved
```

Comment:
This comment isn't accurate now as the change here only queues the message if approved.

EES-1099 wants the ability to cancel a scheduled release and I added a comment to say this could be done by moving the status back to Draft if a BAU user has permission to do this which is EES-957.

Previously this would have made the release service aware of the state change in both directions. The validation cancelled the schedule if it's not approved.